RHEL criterion "is installed" content causes panic

## Description of Problem / Feature Request

Using quay/clair v2 as the base of https://github.com/arminc/clair-local-scan/ (specifically v2.1.6 defined [here](https://github.com/arminc/clair-local-scan/blob/master/.travis.yml#L9)), have discovered very recently that https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml appears to break the following code:

https://github.com/quay/clair/blob/dcf2adb5fef0ec61746812f7930c9eb6923400e4/ext/vulnsrc/rhel/rhel.go#L274-L279

Searching for the `criterion` in the RHEL8.xml file produces:
```
$ grep '<criterion' rhsa-RHEL8.xml | grep 'is installed' | sort -u
      <criterion comment="Red Hat CoreOS 4 is installed" test_ref="oval:com.redhat.rhsa:tst:20211193004"/>
      <criterion comment="Red Hat Enterprise Linux 8 is installed" test_ref="oval:com.redhat.rhba:tst:20191992003"/>
     <criterion comment="Red Hat Enterprise Linux 7 is installed" test_ref="oval:com.redhat.rhsa:tst:20191619011"/>
     <criterion comment="Red Hat Enterprise Linux 8 is installed" test_ref="oval:com.redhat.rhba:tst:20191992003"/>
```

The `"Red Hat CoreOS 4 is installed"` matches the " is installed" check but then obviously does not match `"Red Hat Enterprise Linux "` so the code to extract the `osVersion` via range ultimately panics because of an invalid range: `panic: runtime error: slice bounds out of range [25:24]`.

### Steps to reproduce

```
docker run --name clair-db -d -p 5432:5432 arminc/clair-db:latest
docker run --name clair-server -d --link clair-db:postgres -p 6060-6061 arminc/clair-local-scan:latest
```

### Expected Outcome

That the clair-server container does not panic and exit but completes loading vulnerabilities and waits for scanner (client) connections.

### Actual Outcome

```
...
panic: runtime error: slice bounds out of range [25:24]

goroutine 45 [running]:
github.com/quay/clair/v2/ext/vulnsrc/rhel.toFeatureVersions(0xc00153ca68, 0x2, 0xc0057c3178, 0x1, 0x1, 0xc00aef64d0, 0x1, 0x1, 0x2, 0xc0057c3160, ...)
	/go/clair/ext/vulnsrc/rhel/rhel.go:276 +0xbf8
github.com/quay/clair/v2/ext/vulnsrc/rhel.parseRHSA(0x7f0f8b8c01e8, 0xc00582c000, 0xc00582c000, 0x7f0f8b8c01e8, 0xc00582c000, 0x43, 0x0)
	/go/clair/ext/vulnsrc/rhel/rhel.go:166 +0x1c7
github.com/quay/clair/v2/ext/vulnsrc/rhel.(*updater).Update(0xd053d0, 0x9f24c0, 0xc0000b1340, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/clair/ext/vulnsrc/rhel/rhel.go:133 +0x553
github.com/quay/clair/v2.fetch.func1(0x9f24c0, 0xc0000b1340, 0xc00011c120, 0xc000278300, 0x92a284, 0x4, 0x9e3600, 0xd053d0)
	/go/clair/updater.go:243 +0xa5
created by github.com/quay/clair/v2.fetch
	/go/clair/updater.go:242 +0x1a9
```

### Environment

* Clair version/image: quay.io/coreos/clair:latest
    * Via: https://hub.docker.com/r/arminc/clair-local-scan tag=latest which was built approx 2021-04-18
* Host OS: same result on Ubuntu linux and MacOS (Darwin)

--- a/ext/vulnsrc/rhel/rhel.go
+++ b/ext/vulnsrc/rhel/rhel.go
@@ -269,10 +269,14 @@ func toFeatureVersions(criteria criteria) []database.FeatureVersion {
 			err            error
 		)
 
+		const (
+			rhelPrefix = "Red Hat Enterprise Linux "
+		)
+
 		// Attempt to parse package data from trees of criterions.
 		for _, c := range criterions {
-			if strings.Contains(c.Comment, " is installed") {
-				const prefixLen = len("Red Hat Enterprise Linux ")
+			if strings.Contains(c.Comment, " is installed") && strings.HasPrefix(c.Comment, rhelPrefix) {
+				const prefixLen = len(rhelPrefix)
 				osVersion, err = strconv.Atoi(strings.TrimSpace(c.Comment[prefixLen : prefixLen+strings.Index(c.Comment[prefixLen:], " ")]))
 				if err != nil {
 					log.WithField("criterion comment", c.Comment).Warning("could not parse Red Hat release version from criterion comment")

